### ISSUE TITLE
Downloading large files

### ISSUE THREAD

**Issue by elizoller** [asulibraries/islandora-repo#508]
When a file is of a decent size (lets say 2GB or above), it becomes difficult, if not impossible, to download through the Drupal UI over HTTP. We should consider 
1. an interim solution that just tells people to contact us for the file
2. a longer term solution that perhaps compresses and downloads the file to temp space and then hands it over (thinking about how the large CSV export works for views - it does the work in a "batch" and then presents a link to download?)

open to any other solutions

**Comment by elizoller** [comment]
switching from nginx to apache definitely improved this but didnt fully resolve it

**Comment by seth-shaw-asu** [comment]
Confirmed that this is still an issue. It shows up in the logs with the line `TypeError: stream_copy_to_stream(): Argument #1 ($from) must be of type resource, bool given in stream_copy_to_stream() (line 306 of /var/www/html/drupal/vendor/symfony/http-foundation/BinaryFileResponse.php)`

Nginx gives a bit more information:
```
2022/08/17 20:08:25 [error] 1168#1168: *5415295 FastCGI sent in stderr: "PHP message: PHP Warning:  cURL error 18: transfer closed with 4216221268 bytes remaining to read (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) in /var/www/html/drupal/vendor/twistor/flysystem-stream-wrapper/src/FlysystemStreamWrapper.php on line 932 PHP message: PHP Warning:  fopen(fedora://c2/SchoolOfDance_EmergingArtistResite1_20091106_2.mov): Failed to open stream: &quot;Drupal\flysystem\FlysystemBridge::stream_open&quot; call failed in /var/www/html/drupal/vendor/symfony/http-foundation/BinaryFileResponse.php on line 304PHP message: TypeError: stream_copy_to_stream(): Argument #1 ($from) must be of type resource, bool given in /var/www/html/drupal/vendor/symfony/http-foundation/BinaryFileResponse.php on line 306 #0 /var/www/html/drupal/vendor/symfony/http-foundation/BinaryFileResponse.php(306): stream_copy_to_stream()
#1 /var/www/html/drupal/vendor/symfony/http-foundation/Response.php(381): Symfony\Component\HttpFoundation\BinaryFileResponse->sendContent()
#2 /var/www/html/drupal/web/index.php(20): Symfony\Component\HttpFoundation\Response->send()
#3 {main}" while reading response header from upstream, client: 10.192.19.123, server: keep.lib.asu.edu, request: "GET /_flysystem/fedora/c2/SchoolOfDance_EmergingArtistResite1_20091106_2.mov HTTP/1.1", upstream: "fastcgi://unix:/var/run/php/php-fpm.sock:", host: "keep.lib.asu.edu"
```

**Comment by seth-shaw-asu** [comment]
Fixed by moving files out of Fedora into Drupal private/public.
